tls/s2n_kem.c

```diff
@@ -100,25 +100,23 @@ const struct s2n_iana_to_kem kem_mapping[1] = {
  *
  * https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-8
  * https://datatracker.ietf.org/doc/draft-kwiatkowski-tls-ecdhe-mlkem/
- *
- * The structure of the hybrid share is:
- *    size of ECC key share (2 bytes)
- * || ECC key share (variable bytes)
- * || size of PQ key share (2 bytes)
- * || PQ key share (variable bytes)
  */
 const struct s2n_kem_group s2n_secp256r1_mlkem_768 = {
     .name = "SecP256r1MLKEM768",
     .iana_id = TLS_PQ_KEM_GROUP_ID_SECP256R1_MLKEM_768,
     .curve = &s2n_ecc_curve_secp256r1,
     .kem = &s2n_mlkem_768,
+    .send_kem_first = 0,
 };
 
 const struct s2n_kem_group s2n_x25519_mlkem_768 = {
     .name = "X25519MLKEM768",
     .iana_id = TLS_PQ_KEM_GROUP_ID_X25519_MLKEM_768,
     .curve = &s2n_ecc_curve_x25519,
     .kem = &s2n_mlkem_768,
+    /* ML-KEM KeyShare should always be sent first according to the RFC for X25519MLKEM768.
+     * https://datatracker.ietf.org/doc/html/draft-kwiatkowski-tls-ecdhe-mlkem-02#name-negotiated-groups */
```

Comment:
Would be helpful to pull this document link and a more generic explanation up into the header file that defines this flag.